src/client/opamCliMain.ml

```diff
@@ -206,18 +208,28 @@ let check_and_run_external_commands () =
         then
           let nv =
             try
-              OpamPackage.max_version plugins
-                (OpamPackage.Name.of_string prefixed_name)
+              (* If the command was resolved, attempt to find the package to reinstall. *)
+              if cmd = None then
+                raise Not_found
+              else
+                OpamPackage.package_of_name installed (OpamPackage.Name.of_string prefixed_name)
```

Comment:
Won't this cause problems if the currently installed version is no longer available ? This could happen e.g. with `available:` depending on `opam-version` after an opam root upgrade, where an upgrade of the plugin becomes mandatory, and cause an error during resolution, requiring a manual `opam upgrade`.
I don't remember why this code used to force the latest version (within availability) though

Reply:
It turns out it doesn't, but I think that's a bug - the availability check is not done for `opam reinstall`!

The alternative would be to upgrade the package by name instead?

Reply:
The `candidates` package already have the availability check, so if a plugin is no more available it won't be in `installed` and its latest version will be "newly" installed:
```
$ opam install opam-depext.1.1.3 #without opam < 2.1 condition
$ opam update # reintroduce opam < 2.1 condition
$ rm <link>
$ opam depext
Opam plugin "depext" is not installed. Install it on the current switch? [Y/n] y
opam-depext.1.1.5 is not installed. Install it? [Y/n] y
The following actions will be performed:
  ↗ upgrade opam-depext 1.1.3 to 1.1.5

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⬇ retrieved opam-depext.1.1.5  (cached)
⊘ removed   opam-depext.1.1.3
∗ installed opam-depext.1.1.5
Done.
```

On the availability check for reinstall, this (specific) case is handled by the `orphans` mechanism, and can't be reinstalled:
```
$ opam install opam-depext.1.1.3 #without opam < 2.1 condition
$ opam update # reintroduce opam < 2.1 condition
$ opam reinstall opam-depext
[ERROR] Sorry, these packages are no longer available from the repositories: opam-depext (= 1.1.3)
```